encoding/mvt/layer_internal_test.go

```diff
@@ -0,0 +1,137 @@
+package mvt
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-spatial/geom"
+	"github.com/arolek/p"
+	vectorTile "github.com/go-spatial/geom/encoding/mvt/vector_tile"
+)
+
+func TestLayer(t *testing.T) {
+	type tcase struct {
+		layer   *Layer
+		vtlayer *vectorTile.Tile_Layer
+		eerr    error
+	}
+
+	fn := func(tc tcase) func(t *testing.T) {
+		return func(t *testing.T) {
+			vt, err := tc.layer.VTileLayer(context.Background())
+			if err != tc.eerr {
+				t.Errorf("unexpected error, expected %v got %v", tc.eerr, err)
+			}
+			if tc.vtlayer == nil {
+				if vt != nil {
+					t.Errorf("expected nil value, got non-nil")
+				}
+				return
+			}
+			if vt == nil {
+				t.Errorf("for a Vector Tile, Expected non-nil Got nil")
+				return
+			}
+			if *tc.vtlayer.Version != *vt.Version {
+				t.Errorf("versions do not match, expected %v got %v", *tc.vtlayer.Version, *vt.Version)
+			}
+			if *tc.vtlayer.Name != *vt.Name {
+				t.Errorf("names do not match, expected %v got %v", *tc.vtlayer.Name, *vt.Name)
+			}
+			if *tc.vtlayer.Extent != *vt.Extent {
+				t.Errorf("extent do not match, expected %v got %v", *tc.vtlayer.Extent, *vt.Extent)
+			}
+			if len(tc.vtlayer.Features) != len(vt.Features) {
+				t.Errorf("features do not have the same length, expected %v got %v", len(tc.vtlayer.Features), len(vt.Features))
+			}
+
+			// TODO: Should check to see if the features are equal.
+			if len(tc.vtlayer.Values) != len(vt.Values) {
+				t.Errorf("values do not have the same length, expected %v got %v", len(tc.vtlayer.Values), len(vt.Values))
+			}
+
+			// TODO: Should check that the Values are equal.
+		}
+	}
+
+	tests := map[string]tcase{
+		"no features": {
+			layer: &Layer{
+				Name: "nofeatures",
+			},
+			vtlayer: &vectorTile.Tile_Layer{
+				Version:  p.Uint32(Version),
+				Name:     p.String("nofeatures"),
```

Comment:
Move Version from cons block above into a vars block above. And then just use `&Version` here. This is causing an allocation for no good reason. Same with `nofeatures` Make it a package global and then take a reference to it. Same with `DefaultExtent`.